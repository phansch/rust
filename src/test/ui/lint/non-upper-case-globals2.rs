#![allow(dead_code)]
#![deny(non_upper_case_globals)]

static foo: isize = 1;
static mut bar: isize = 1;

fn main() {
    const b: usize = 1;
}

trait Foo {
    const camelCase: usize;
}

impl Foo for i32 {
    const camelCase: usize = 2;
}
