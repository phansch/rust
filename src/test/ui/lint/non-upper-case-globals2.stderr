error: static variable `foo` should have an upper case name such as `FOO`
 --> $DIR/non-upper-case-globals2.rs:4:8
  |
4 | static foo: isize = 1;
  |        ^^^
  |
note: lint level defined here
 --> $DIR/non-upper-case-globals2.rs:2:9
  |
2 | #![deny(non_upper_case_globals)]
  |         ^^^^^^^^^^^^^^^^^^^^^^

error: static variable `bar` should have an upper case name such as `BAR`
 --> $DIR/non-upper-case-globals2.rs:5:12
  |
5 | static mut bar: isize = 1;
  |            ^^^

error: constant `b` should have an upper case name such as `B`
 --> $DIR/non-upper-case-globals2.rs:8:11
  |
8 |     const b: usize = 1;
  |           ^

error: associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
  --> $DIR/non-upper-case-globals2.rs:12:11
   |
12 |     const camelCase: usize;
   |           ^^^^^^^^^

error: associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
  --> $DIR/non-upper-case-globals2.rs:16:11
   |
16 |     const camelCase: usize = 2;
   |           ^^^^^^^^^

error: aborting due to 5 previous errors

